Loop/Managers/LoopDataManager.swift

```diff
@@ -278,12 +386,40 @@ class LoopDataManager {
         }
     }
 
+    private func updateIOB(completionHandler: (values: [InsulinValue]?, error: ErrorType?) -> Void) {
+        let glucose = deviceDataManager.glucoseStore?.latestGlucose
+
+        deviceDataManager.doseStore.getInsulinOnBoardValues(startDate: glucose?.startDate) { (values, error) -> Void in
```

Comment:
Since you only need one value here, use `deviceDataManager.doseStore.insulinOnBoardAtDate(startDate)`